bot/exts/fun/uwu.py

```diff
@@ -116,10 +61,57 @@ class Uwu(Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
+    def _word_replace(self, input_string: str) -> str:
+        """Replaces words that are keys in the word replacement hash to the values specified."""
+        for word in WORD_REPLACE_HASH:
+            input_string = input_string.replace(word, WORD_REPLACE_HASH[word])
+        return input_string
+
+    def _char_replace(self, input_string: str) -> str:
+        """Replaces characters with 'w'."""
+        return REGEX_WORD_REPLACE.sub("w", input_string)
+
+    def _stutter(self, strength: float, input_string: str) -> str:
+        """Adds stuttering to a string."""
+        return REGEX_STUTTER.sub(partial(self._stutter_replace, strength=strength), input_string, 0)
+
+    def _stutter_replace(self, match: Callable, strength: float = 0.0) -> str:
+        """Replaces a single character with a stuttered character."""
+        match_string = match.string[slice(*match.span())]
+        if random.random() < strength:
+            char = match_string[-1]
+            return f"{match_string}-{char}"
+        return match_string
+
+    def _nyaify(self, input_string: str) -> str:
+        """Nyaifies a string."""
+        return REGEX_NYA.sub(SUBSTITUTE_NYA, input_string, 0)
+
+    def _emoji(self, strength: float, input_string: str) -> str:
+        """Replaces some punctuation with emoticons."""
+        return REGEX_PUNCTUATION.sub(partial(self._emoji_replace, strength=strength), input_string, 0)
+
+    def _emoji_replace(self, match: Callable, strength: float = 0.0) -> str:
+        """Replaces a punctuation character with an emoticon."""
+        match_string = match.string[slice(*match.span())]
```

Comment:
This can be simplified.
```suggestion
        match_string = match.group()
```